CMakeLists.txt

```diff
@@ -594,6 +594,30 @@ include_directories(SYSTEM ${PROJECT_BINARY_DIR}/include)
 
 find_package(Threads REQUIRED)
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
```

Comment:
@adamemerson thanks for getting this right. i am just curious why you choosed to check the output of c++ compiler instead of using `_LIBCPP_VERSION`  and `__GLIBCXX__` in the `librarycheck.cc` file?

Reply:
also, if we use the combination of gcc and libc++, shall we still link against `c++experimental` as well?

Reply:
@tchaikov I think no?

Reply:
@mattbenjamin why? as i don't think `stdc++fs` should be linked if libc++ is used.